main/src/http/http_service.rs

```diff
@@ -160,42 +160,83 @@ impl HttpService {
-        self.ping()
-            .or(self.query())
-            .or(self.write_line_protocol())
-            .or(self.metrics())
-            .or(self.print_meta())
-            .or(self.debug_pprof())
-            .or(self.debug_jeprof())
-            .or(self.prom_remote_read())
-            .or(self.prom_remote_write())
-            .or(self.backtrace())
+        #[cfg(unix)]
+        {
+            self.ping()
+                .or(self.query())
+                .or(self.write_line_protocol())
+                .or(self.metrics())
+                .or(self.print_meta())
+                .or(self.prom_remote_read())
+                .or(self.prom_remote_write())
+                .or(self.backtrace())
+                .or(self.debug_pprof())
+                .or(self.debug_jeprof())
+        }
+
+        #[cfg(not(unix))]
+        {
+            self.ping()
+                .or(self.query())
+                .or(self.write_line_protocol())
+                .or(self.metrics())
+                .or(self.print_meta())
+                .or(self.prom_remote_read())
+                .or(self.prom_remote_write())
+                .or(self.backtrace())
+        }
     }
 
     fn routes_query(
         &self,
     ) -> impl Filter<Extract = (impl warp::Reply,), Error = warp::Rejection> + Clone {
-        self.ping()
-            .or(self.query())
-            .or(self.metrics())
-            .or(self.print_meta())
-            .or(self.debug_pprof())
-            .or(self.debug_jeprof())
-            .or(self.prom_remote_read())
-            .or(self.backtrace())
+        #[cfg(unix)]
+        {
+            self.ping()
+                .or(self.query())
+                .or(self.metrics())
+                .or(self.print_meta())
+                .or(self.prom_remote_read())
+                .or(self.backtrace())
+                .or(self.debug_pprof())
+                .or(self.debug_jeprof())
+        }
+
+        #[cfg(not(unix))]
+        {
+            self.ping()
+                .or(self.query())
+                .or(self.metrics())
+                .or(self.print_meta())
+                .or(self.prom_remote_read())
+                .or(self.backtrace())
+        }
     }
 
     fn routes_store(
         &self,
     ) -> impl Filter<Extract = (impl warp::Reply,), Error = warp::Rejection> + Clone {
-        self.ping()
-            .or(self.write_line_protocol())
-            .or(self.metrics())
-            .or(self.print_meta())
-            .or(self.debug_pprof())
-            .or(self.debug_jeprof())
-            .or(self.prom_remote_write())
-            .or(self.backtrace())
+        #[cfg(unix)]
+        {
+            self.ping()
+                .or(self.write_line_protocol())
+                .or(self.metrics())
+                .or(self.print_meta())
+                .or(self.prom_remote_write())
+                .or(self.backtrace())
+                .or(self.debug_pprof())
+                .or(self.debug_jeprof())
+        }
+
```

Comment:
Mask the operating system inside the function

Reply:
Resolved